AssetRipper.GUI/ConsoleApp.cs

```diff
@@ -1,126 +1,179 @@
+				object? settingValue = context.ParseResult.GetValueForOption(option);
 
-				return false;
-			}
+				if (settingValue == null)
+					continue;
 
-			try
-			{
-				options.LogFile ??= new FileInfo(ExecutingDirectory.Combine(DefaultLogFileName));
-				options.OutputDirectory ??= new DirectoryInfo(ExecutingDirectory.Combine("Ripped"));
-			}
-			catch (Exception ex)
-			{
-				System.Console.WriteLine($"Failed to initialize the output and log paths.");
-				System.Console.WriteLine(ex.ToString());
-				return false;
+				string? enumKey = settingValue.ToString();
+
+				if (string.IsNullOrEmpty(enumKey))
+					continue;
+
+				ripper.Settings.SetSetting(type, Enum.Parse(type, enumKey));
 			}
 
-			return true;
+			DoRipping(ripper, inputs, output.FullName, verbose, logFile.FullName);
 		}
 
-		private static void Run(Options options)
+		private static void HandleException(Exception ex, InvocationContext context)
 		{
-			Logger.AllowVerbose = options.Verbose;
-			Logger.Add(new ConsoleLogger(false));
-			Logger.Add(new FileLogger(options.LogFile.FullName));
-			Logger.LogSystemInformation("AssetRipper Console Version");
-#if !DEBUG
-			try
-#endif
-			{
-				Ripper ripper = new();
-				ripper.Settings.LogConfigurationValues();
-				ripper.Load(options.FilesToExport);
-				PrepareExportDirectory(options.OutputDirectory.FullName);
-				ripper.ExportProject(options.OutputDirectory.FullName);
-			}
-#if !DEBUG
-			catch (Exception ex)
-			{
-				Logger.Log(LogType.Error, LogCategory.General, ex.ToString());
-			}
+			HandleException(ex);
+		}
+
+		private static void HandleException(Exception ex)
+		{
+			Environment.ExitCode = (int)ExitCode.ExtractingError;
+
+#if DEBUG
+			Console.WriteLine("===================================================");
+			Console.WriteLine("      Error during ripping, extraction aborted     ");
+			Console.WriteLine("===================================================");
+			Console.WriteLine(ex.ToString());
+#else
+			Console.WriteLine("Error during ripping, extraction aborted");
+			Console.WriteLine("See this message below for details:");
+			Console.WriteLine(ex.Message);
 #endif
 		}
 
-		private static void PrepareExportDirectory(string path)
+		public static void ParseArgumentsAndRun(string[] args)
 		{
-			if (Directory.Exists(path))
+			Ripper ripper = new();
+			ripper.Settings.ResetToDefaultValues();
+
+			RootCommand rootCommand = new RootCommand();
+			rootCommand.SetHandler((InvocationContext context) => RootCommandHandler(context, ripper));
+			rootCommand.AddArgument(ConsoleOptions.inputArgument);
+			rootCommand.AddOption(ConsoleOptions.outputOption);
+			rootCommand.AddOption(ConsoleOptions.quitOption);
+			rootCommand.AddOption(ConsoleOptions.verboseOption);
+			rootCommand.AddOption(ConsoleOptions.logFileOption);
+
+			Dictionary<string, Option> options = ConsoleOptions.GenerateFromRipper(ripper);
+
+			Command extractCommand = new Command("extract");
```

Comment:
I am opposed to any command that doesn't start with `-`.

Reply:
This is how System.CommandLine works, you have a root command, and then subcommands, like in dotnet for example you have `dotnet build` and `dotnet restore`, that each have different options.  

https://docs.microsoft.com/en-us/dotnet/standard/commandline/syntax#subcommands